src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -548,39 +574,45 @@ private void ParseExtension(ReadableBuffer buffer, out ReadCursor consumed, out
                 // Just drain the data
                 consumed = buffer.Start;
                 examined = buffer.Start;
+
                 do
                 {
                     ReadCursor extensionCursor;
                     if (ReadCursorOperations.Seek(buffer.Start, buffer.End, out extensionCursor, ByteCR) == -1)
                     {
                         // End marker not found yet
+                        CheckExaminedBytes(buffer.Length);
```

Comment:
It's super trivial to change this to just count the "data" bytes if that's what we decide. I decided to do the more difficult thing first.

I'm not too concerned about breaking any apps by counting extension bytes towards the request body size limit since chunk extensions are so rare. Counting the framing bytes might make Kestrel behave differently that HttpSysServer, but the framing can easily take up to ~92% of the uploaded bytes without any chunk extensions or trailing headers.  We can discuss offline how much we care about that.